pkg/signing/signing.go

```diff
@@ -223,6 +367,70 @@ func SifAddSignature(sinfo *image.Sifinfo, signature []byte) error {
 	return nil
 }
 
+func sykeysFetchPubkey(fingerprint string, sykeysAddr string) (openpgp.EntityList, error) {
+	url := "http://" + sykeysAddr + "/pks/lookup?op=get&options=mr&search=0x" + fingerprint
+	log.Println("url :", url)
+	res, err := http.Get(url)
+	if err != nil {
+		log.Println(err)
+		return nil, err
+	}
+
+	el, err := openpgp.ReadArmoredKeyRing(res.Body)
+	res.Body.Close()
+	if err != nil {
+		log.Println("error while trying to read armored key ring: ", err)
+		return nil, err
+	}
+	if len(el) == 0 {
+		err = fmt.Errorf("no keys in keyring")
+		log.Println(err)
+		return nil, err
+	}
+	if len(el) > 1 {
+		err = fmt.Errorf("server returned more than one key for unique fingerprint")
+		log.Println(err)
+		return nil, err
+	}
+
+	return el, nil
+}
+
+func sykeysPushPubkey(entity *openpgp.Entity, sykeysAddr string) (err error) {
+	u := "http://" + sykeysAddr + "/pks/add"
+	w := bytes.NewBuffer(nil)
+
+	wr, err := armor.Encode(w, openpgp.PublicKeyType, nil)
+	if err != nil {
+		log.Println("armor.Encode failed")
+	}
+
+	err = entity.Serialize(wr)
+	if err != nil {
+		log.Println("can't serialize public key", err)
+		return err
+	}
+	wr.Close()
+
+	d := url.Values{}
+	d.Set("keytext", w.String())
+	resp, err := http.PostForm(u, d)
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+
+	text, err := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+	fmt.Printf("%s", text)
```

Comment:
I think you can replace this entire section (starting with the `ioutil.ReadAll` line) with:

```
defer resp.Body.Close()

if _, err := io.Copy(os.Stdout, r.Body); err != nil {
	log.Println(err)
	return err
}
```

Reply:
Maybe, will try it !